africanus/rime/fast_beam_cubes.py

```diff
@@ -134,9 +112,18 @@ def beam_cube_dde(beam, beam_lm_extents, beam_freq_map,
                 l, m = lm[s]
 
                 for f in range(nchan):
+                    # Unpack frequency data
+                    freq_scale = freq_data[f, 0]
+                    # lower and upper frequency weights
+                    nud = freq_data[f, 1]
+                    inv_nud = 1.0 - nud
+                    # lower and upper frequency grid position
+                    gc0 = np.int32(freq_data[f, 2])
+                    gc1 = gc0 + 1
+
                     # Apply any frequency scaling
-                    sl = l * freq_scale[f]
-                    sm = m * freq_scale[f]
+                    sl = l * freq_scale
+                    sm = m * freq_scale
```

Comment:
@JoshVStaden I would add print statements here to make sure that the l and m coordinates are the same as those in your Zernike DDEs.